ui/perfherder/compare/CompareTable.jsx

```diff
@@ -28,24 +44,212 @@ export default class CompareTable extends React.PureComponent {
       displayNumber(percentage),
     )}% ${improvement ? 'better' : 'worse'})`;
 
+  compareBy = (key, shouldBeNumber) => {
+    const { sortOrderAsc } = this.state;
+    return (a, b) => {
+      // If a[key] or b[key] doesn't exist, they are skipped by the sorting
+      // so they need to be initialized with a default value
+      if (shouldBeNumber) {
+        // eslint-disable-next-line no-restricted-globals
+        if (isNaN(a[key]) && !Array.isArray(a[key])) a[key] = 0;
+        // eslint-disable-next-line no-restricted-globals
+        if (isNaN(b[key]) && !Array.isArray(b[key])) b[key] = 0;
```

Comment:
I think this function could be simplified. You can create a custom sort function rather than later calling sort with it, like here: https://github.com/mozilla/treeherder/blob/master/ui/intermittent-failures/helpers.js#L77

That `sortData` function in the link above is not being used, so if you find it helpful you can refactor it and move it to a general helpers file  in `ui/helpers`.

I'm curious why the `no-restricted-globals` rule is failing here. We try to disable rules only when it's really necessary.

Reply:
I found only `numericCompare` in `perfherder/helpers.js`. I might need to refactor that to work also for array lengths, if that's ok.
I'm not sure what `no-restricted-globals` means exactly. I'll research more of it and try to make it work without disabling it.

Reply:
I have a better form of this function. I'm going to make a PR to get your feedback and then move it to `ui/helpers`